devtools/scumm-md5.txt

```diff
@@ -39,13 +39,14 @@
 # of detect files, automatically inserting the values in here:
 #
 # #!/bin/sh
+# set -f
 # OIFS=$IFS
 # IFS="
 # "
-# for i in `find -name 00.*` `find -name 000.*` `find -name *.??0` `find -name *Data*` ; do
-#	m=`head -c 1048576 "$i" | md5sum | cut -f1 -d' '` ;
-#	s=`du -b "$i" | cut -f1` ;
-#	sed -i -e "s/$m\t-1/$m\t$s/" /PATH/TO/scumm-md5.txt ;
+# for i in $(find . \( -name '00.*' -o -name '000.*' -o -name '*.??0' -o -name '*Data*' \) -type f); do
```

Comment:
Instead of mangling `IFS` and using `-f`, you can replace the `for` by:
```
find . \( -name '00.*' -o -name '000.*' -o -name '*.??0' -o -name '*Data*' \) -type f | while read i; do
...
done
```
This reads find output line by line without any expansion.

Reply:
Sure! The lesser the better, thanks. I've used `read -r` though, so that it also disables any backslash escaping.

I'm not sure this loop would work if one has newline characters in their filenames, but oh well. This script is meant for the realistic file names of a developper, and I've checked that it still works with the stuff we really care about, such as `Full Throttle Data` files.